src/AppBundle/Resources/public/js/contacts.js

```diff
@@ -50,6 +60,89 @@ $(function () {
 
             if (show) {
                 $(this).fadeIn();
+                $("#contacts").show();
+            } else {
+                $(this).fadeOut();
+            }
+        });
+    }
+
+    function filterGlobalContacts () {
+
+        $("#global_contacts").hide();
+
+        $('#global_contacts .contact').each(function () {
+            var contactGoals = $(this).data('goals') || {};
+            var show = false;
+
+            if (selectedGoals.length !== 0) {
+                for (var key in selectedGoals) {
+                    var selectedGoal = selectedGoals[key];
+
+                    if (contactGoals.hasOwnProperty(selectedGoal)) {
+                        if (selectedCities.length === 0) {
+                            //user want this goal in all cities
+                            show = true;
+                        }
+
+                        if (contactGoals[selectedGoal].length === 0) {
+                            //contact performs this goal in all cities
+                            show = true;
+                        }
+
+                        if (selectedCities.intersects(contactGoals[selectedGoal])) {
+                            //user selected both goal and city, and they intersect with contact
+                            show = true;
```

Comment:
@daria-sieroshtan 
looks like there is A LOT of copy-paste in this file. Copy-paste is an evil offspring of Satan. We cannot let it be.
Please find a way to avoid it, either by extracting reusable part to separate function or by combining all 3 functions together